comp/api/api/utils/common.go

```diff
@@ -0,0 +1,27 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2023-present Datadog, Inc.
+
+// Package utils has common utility methods that components can use for structuring http responses of their endpoints
+package utils
+
+import (
+	"encoding/json"
+	"net"
+	"net/http"
+
+	"github.com/DataDog/datadog-agent/pkg/util/grpc"
+)
+
+// SetJSONError writes a server error as JSON with the correct http error code
+func SetJSONError(w http.ResponseWriter, err error, errorCode int) {
+	w.Header().Set("Content-Type", "application/json")
+	body, _ := json.Marshal(map[string]string{"error": err.Error()})
+	http.Error(w, string(body), errorCode)
+}
+
+// GetConnection returns the connection for the request
+func GetConnection(r *http.Request) net.Conn {
+	return r.Context().Value(grpc.ConnContextKey).(net.Conn)
+}
```

Comment:
Not for this PR, but we must make a note that we should try separate grpc helper functions from regular helper functions. for the main agent API grpc is a given, but it won't be the case for all agents